lighthouse-core/gather/fetcher.js

```diff
@@ -84,19 +88,105 @@ class Fetcher {
   }
 
   /**
-   * Requires that `driver.enableRequestInterception` has been called.
+   * Requires that `fetcher.enable` has been called.
    *
    * Fetches any resource in a way that circumvents CORS.
    *
    * @param {string} url
-   * @param {{timeout: number}} options timeout is in ms
+   * @param {{timeout: number}=} options timeout is in ms
    * @return {Promise<string>}
    */
-  async fetchResource(url, {timeout = 500}) {
+  async fetchResource(url, options = {timeout: 500}) {
     if (!this._enabled) {
-      throw new Error('Must call `enableRequestInterception` before using fetchResource');
+      throw new Error('Must call `enable` before using fetchResource');
+    }
+
+    const milestone = await this.driver.getBrowserVersion().then(v => v.milestone);
+    if (milestone >= 88) {
+      return await this._fetchResourceOverProtocol(url, options);
+    }
+    return await this._fetchResourceIframe(url, options);
+  }
+
+  /**
+   * @param {string} handle
+   * @param {{timeout: number}=} options,
+   * @return {Promise<string>}
+   */
+  async _readIOStream(handle, options = {timeout: 500}) {
+    const startTime = Date.now();
+
+    let ioResponse;
+    let data = '';
+    while (!ioResponse || !ioResponse.eof) {
+      const elapsedTime = Date.now() - startTime;
+      if (elapsedTime > options.timeout) {
+        throw new Error('Waiting for the end of the IO stream exceeded the allotted time.');
+      }
+      ioResponse = await this.driver.sendCommand('IO.read', {handle});
+      const responseData = ioResponse.base64Encoded ?
+        Buffer.from(ioResponse.data, 'base64').toString('utf-8') :
+        ioResponse.data;
+      data = data.concat(responseData);
+    }
+
+    return data;
+  }
+
+  /**
+   * @param {string} url
+   */
+  async _fetchStream(url) {
```

Comment:
nit: this name is a little confusing in relation to `_readIOStream` (here we aren't fetching a stream, we're fetching...via a stream?) and because it's not returning a stream in the node/web streams sense, but in a (string)-handle-to-a-devtools-stream sense.

`_fetchAsStream` or something? Or `_loadNetworkResource ` to match the protocol method and let the return type handle the "stream" part.